subaccounts.go

```diff
@@ -44,30 +49,30 @@ func (c *Client) SubaccountCreate(s *Subaccount) (res *Response, err error) {
 func (c *Client) SubaccountCreateContext(ctx context.Context, s *Subaccount) (res *Response, err error) {
 	// enforce required parameters
 	if s == nil {
```

Comment:
I am generally of the opinions these types of errors/enforcements should be handled by the server and not the client library. If the rule changes on the server it should not require an update to the client library.

If they must be enforced in the client library they should be on the setters and not on the serialization call (fail fast)